pkg/agent/agent.go

```diff
@@ -136,16 +150,34 @@ func (k *Klocksmith) process(stop <-chan struct{}) error {
 		return err
 	}
 
+	glog.Info("Checking if node is already unschedulable")
+	node, err = k8sutil.GetNodeRetry(k.nc, k.node)
+	if err != nil {
+		return err
+	}
+	alreadyUnschedulable := node.Spec.Unschedulable
+
 	// drain self equates to:
-	// 1. set Unschedulable
+	// 1. set Unschedulable if necessary
 	// 2. delete all pods
 	// unlike `kubectl drain`, we do not care about emptyDir or orphan pods
 	// ('any pods that are neither mirror pods nor managed by
 	// ReplicationController, ReplicaSet, DaemonSet or Job')
 
-	glog.Info("Marking node as unschedulable")
-	if err := k8sutil.Unschedulable(k.nc, k.node, true); err != nil {
-		return err
+	if alreadyUnschedulable == false {
```

Comment:
`if !alreadyUnschedulable`

Reply:
Is marking as unschedulable idempotent? If so this check is unnecessary.

Reply:
It is idempotent. Whether we mark the node as unschedulable isn't the important part, it's setting the annotation that indicates the agent is the one that made the node unschedulable.